osbs/api.py

```diff
@@ -10,11 +10,25 @@
 from osbs.build import BuildManager, BuildResponse
 from osbs.constants import DEFAULT_NAMESPACE
 from osbs.core import Openshift
-from osbs.exceptions import OsbsResponseException
+from osbs.exceptions import OsbsException, OsbsResponseException
 
 
+def osbsapi(func):
+    def catch_exceptions(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except OsbsException:
+            # Re-raise OsbsExceptions
+            raise
+        except Exception as e:
+            # Convert anything else to OsbsException
+            raise OsbsException(e)
```

Comment:
Sure, we can do that. In python3, of course, `__context__` makes it clear what the original exception was (and the traceback shows it).